src/openlifu/io/LIFUUart.py

```diff
@@ -0,0 +1,345 @@
+import asyncio
+import logging
+import threading
+import time
+
+import serial
+import serial.tools.list_ports
+
+from openlifu.io.config import (
+    OW_ACK,
+    OW_BAD_PARSE,
+    OW_CMD_NOP,
+    OW_END_BYTE,
+    OW_START_BYTE,
+)
+from openlifu.io.LIFUSignal import LIFUSignal
+from openlifu.io.uartpacket import UartPacket
+from openlifu.io.utils import util_crc16
+
+# Set up logging
+logging.basicConfig(level=logging.DEBUG)
+log = logging.getLogger("UART")
+
+
+class LIFUUart:
+    def __init__(self, vid, pid, baudrate=921600, timeout=10, align=0, demo_mode=False):
+        """
+        Initialize the UART instance.
+
+        Args:
+            vid (int): Vendor ID of the USB device.
+            pid (int): Product ID of the USB device.
+            baudrate (int): Communication speed.
+            timeout (int): Read timeout in seconds.
+            align (int): Data alignment parameter.
+        """
+        self.vid = vid
+        self.pid = pid
+        self.port = None
+        self.baudrate = baudrate
+        self.timeout = timeout
+        self.align = align
+        self.packet_count = 0
+        self.serial = None
+        self.running = False
+        self.asyncMode = False
+        self.loop = asyncio.get_event_loop()
```

Comment:
@georgevigelette Currently if I try to instantiate a `LIFUInterface` I get an error at this line:

```txt
RuntimeError: There is no current event loop in thread 'MainThread'.
```

This happens whether I set `test_mode=True` or not